wavefront_python_sdk/common/utils.py

```diff
@@ -0,0 +1,241 @@
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import re
+import io
+from gzip import GzipFile
+import threading
+
+
+class AtomicCounter(object):
+    """
+    An atomic, thread-safe incrementing counter.
+    """
+
+    def __init__(self, initial=0):
+        self.value = initial
+        self._lock = threading.Lock()
+
+    def increment(self, num=1):
+        """
+        Increment atomic counter value
+        @param num: Num to be increased, 1 by default
+        @return: Current value after increment
+        """
+        with self._lock:
+            self.value += num
+            return self.value
+
+
+def chunks(data_list, batch_size):
+    """
+    Split list of data into chunks with fixed batch size
+    @param data_list: List of data
+    @param batch_size: Batch size of each chunk
+    @return: Return a lazy generator object for iteration
+    """
+    for i in range(0, len(data_list), batch_size):
+        yield data_list[i:i + batch_size]
+
+
+def gzip_compress(data, compresslevel=9):
+    """
+    Compress data using GZIP
+    @param data: Data to compress
+    @param compresslevel: Compress Level
+    @return: Compressed data
+    """
+    buf = io.BytesIO()
+    with GzipFile(fileobj=buf, mode='wb', compresslevel=compresslevel) \
+            as gzip_file:
+        gzip_file.write(data)
+    return buf.getvalue()
+
+
+def sanitize(string):
+    """
+    Sanitize a string, replace whitespace with "-"
+    @param string: Input string
+    @return: Sanitized string
+    """
+    whitespace_sanitized = re.sub(r"[\s]+", "-", string)
+    if "\"" in whitespace_sanitized:
+        return "\"" + re.sub(r"[\"]+", '\\\\\"', whitespace_sanitized) + "\""
+    else:
+        return "\"" + whitespace_sanitized + "\""
+
+
+def is_blank(string):
+    """
+    Check is a string is black or not, either none or only contains whitespace
+    @param string: String to be checked
+    @return: Is blank or not
+    """
+    return string is None or len(string) == 0 or string.isspace()
+    # return len(re.sub(r"[\s]+", "", s)) == 0
+
+
+def metric_to_line_data(name, value, timestamp, source, tags, default_source):
+    """
+    Metric Data to String
+    Wavefront Metrics Data format
+    <metricName> <metricValue> [<timestamp>] source=<source> [pointTags]
+    Example: "new-york.power.usage 42422 1533531013 source=localhost
+              datacenter=dc1"
+    @param name: Metric Name
+    @type name: str
+    @param value: Metric Value
+    @type value: float
+    @param timestamp: Timestamp
+    @type timestamp: long
+    @param source: Source
+    @type source: str
+    @param tags: Tags
+    @type tags: dict
+    @param default_source:
+    @type default_source: str
+    @return: String
+    """
+    if is_blank(name):
+        raise ValueError("Metrics name cannot be blank")
+
+    if is_blank(source):
+        source = default_source
+
+    str_builder = [sanitize(name), float(value)]
```

Comment:
Float here needs to be converted into a string, not to break the string concatenation later.